pkg/controller/jobs/kubeflow/common/common.go

```diff
@@ -0,0 +1,89 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+)
+
+type objAsPtr[T any] interface {
+	metav1.Object
+	client.Object
+	*T
+}
+
+func SyncJob[PtrT objAsPtr[T], T any](
```

Comment:
why not put this in a generic struct that implements multikueue adapter and has two members holding the specialized "Update..." functions 

then the job will just need to define the two functions and call a `MakeAdaptor(f1, f2)`.